Samples/kotlin-device-support/src/main/java/com/phunware/kotlin/sample/sensors/location/BluedotLocationActivity.kt

```diff
@@ -0,0 +1,186 @@
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+Except as contained in this notice, the name of Phunware, Inc. shall
+not be used in advertising or otherwise to promote the sale, use or
+other dealings in this Software without prior written authorization
+from Phunware, Inc. */
+
+import android.content.Context
+import android.hardware.Sensor
+import android.hardware.SensorManager
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.*
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.model.MapStyleOptions
+import com.phunware.core.PwCoreSession
+import com.phunware.core.PwLog
+import com.phunware.kotlin.sample.sensors.CustomMapFragment.Companion.newInstance
+import com.phunware.kotlin.sample.sensors.R
+import com.phunware.kotlin.sample.sensors.building.adapter.FloorAdapter
+import com.phunware.location.provider_managed.ManagedProviderFactory
+import com.phunware.location.provider_managed.PwManagedLocationProvider
+import com.phunware.mapping.MapFragment
+import com.phunware.mapping.OnPhunwareMapReadyCallback
+import com.phunware.mapping.PhunwareMap
+import com.phunware.mapping.manager.Callback
+import com.phunware.mapping.manager.PhunwareMapManager
+import com.phunware.mapping.model.Building
+import com.phunware.mapping.model.FloorOptions
+import java.lang.ref.WeakReference
+
+class BluedotLocationActivity : AppCompatActivity(), OnPhunwareMapReadyCallback,
+        Building.OnFloorChangedListener {
+
+    private lateinit var mapManager: PhunwareMapManager
+    private lateinit var mapFragment: MapFragment
+    private lateinit var currentBuilding: Building
+    private lateinit var floorSpinner: Spinner
+    private lateinit var floorSpinnerAdapter: ArrayAdapter<FloorOptions>
+    private lateinit var content: RelativeLayout
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_bluedot_location)
+        content = findViewById(R.id.content)
+
+        floorSpinner = findViewById(R.id.floorSpinner)
+        floorSpinnerAdapter = FloorAdapter(this)
+        floorSpinner.adapter = floorSpinnerAdapter
+        floorSpinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
+                val floor = floorSpinnerAdapter.getItem(id.toInt())
+                if (floor != null) {
+                    currentBuilding.selectFloor(floor.level)
+                }
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>) {}
+        }
+
+        // Register the Phunware API keys
+        PwCoreSession.getInstance().environment = PwCoreSession.Environment.STAGE
```

Comment:
Did we want to keep this always pointed to stage?

Reply:
no. great catch. removed.